src/autocomplete/powershell.ts

```diff
@@ -0,0 +1,455 @@
+              ...genNode(`${partialId}:${topicNameSplit[depth]}`),
+            }
+          }
+        }
+      }
+
+      for (const c of this.commands) {
+        const cmdIdSplit = c.id.split(':')
+
+        if (partialId === c.id && this.coTopics.includes(c.id)) {
+          node._command = c.id
+        }
+
+        if (
+          c.id.startsWith(partialId + ':') &&
+          cmdIdSplit.length === depth + 1 &&
+          !nextArgs.includes(cmdIdSplit[depth])
+        ) {
+          node[cmdIdSplit[depth]] = {
+            _command: c.id,
+          }
+        }
+      }
+      return node
+    }
+
+    const commandTree: Record<string, any> = {}
+
+    const topLevelArgs: string[] = []
+
+    // Collect top-level topics and generate a cmd tree node for each one of them.
+    this.topics.forEach(t => {
+      if (!t.name.includes(':')) {
+        if (this.coTopics.includes(t.name)) {
+          commandTree[t.name] = {
+            ...genNode(t.name),
+          }
+        } else {
+          commandTree[t.name] = {
+            _summary: t.description,
+            ...genNode(t.name),
+          }
+        }
+
+        topLevelArgs.push(t.name)
+      }
+    })
+
+    // Collect top-level commands and add a cmd tree node with the command ID.
+    this.commands.forEach(c => {
+      if (!c.id.includes(':') && !this.coTopics.includes(c.id)) {
+        commandTree[c.id] = {
+          _command: c.id,
+        }
+
+        topLevelArgs.push(c.id)
+      }
+    })
+
+    const hashtables: string[] = []
+
+    for (const topLevelArg of topLevelArgs) {
+      // Generate all the hashtables for each child node of a top-level arg.
+      hashtables.push(this.genHashtable(topLevelArg, commandTree))
+    }
+
+    const commandsHashtable = `
+@{
+${hashtables.join('\n')}
+}`
+
+    const compRegister = `
+using namespace System.Management.Automation
+using namespace System.Management.Automation.Language
+
+$scriptblock = {
+    param($WordToComplete, $CommandAst, $CursorPosition)
+
+    $Commands = ${commandsHashtable}
+
+    # Get the current mode
+    $Mode = (Get-PSReadLineKeyHandler | Where-Object {$_.Key -eq "Tab" }).Function
+
+    # Everything in the current line except the CLI executable name.
+    $CurrentLine = $commandAst.CommandElements[1..$commandAst.CommandElements.Count] -split " "
+
+    # Save flags in current line without the \`--\` prefix.
+    $Flags = $CurrentLine | Where-Object {
+      $_ -Match "^-{1,2}(\\w+)"
+    } | ForEach-Object {
+      $_.trim("-")
+    }
+    # Set $flags to an empty hashtable if there are no flags in the current line.
+    if ($Flags -eq $null) {
+      $Flags = @{}
+    }
+
+    # No command in the current line, suggest top-level args.
+    if ($CurrentLine.Count -eq 0) {
+        $Commands.GetEnumerator() | Sort-Object -Property key | ForEach-Object {
+          New-Object -Type CompletionResult -ArgumentList \`
+              $($Mode -eq "MenuComplete" ? "$($_.Key) " : "$($_.Key)"),
```

Comment:
add a space at the end of the suggestion if mode is `MenuComplete`.

In MenuComplete mode you have to press <Enter> to select a suggestion, without the additional space at the end the user would need to press <Space> and <Tab> again after each completion.
![Screenshot 2023-04-19 at 12 44 32](https://user-images.githubusercontent.com/6853656/233129550-5770b4f0-7710-40f2-b066-4e839b5b2461.png)